### ISSUE TITLE
Abstract library to accept alternative JXM providers

### ISSUE THREAD

**Issue by Swivelgames** [repetere/jsonx#671]
It's not a major change, but certainly a major paradigm pivot. It would be awesome to pivot this library from being strictly React specific, and allow the user to specify a JXM renderer themselves.

i.e., instead of using `React.createElement`, it would be cool if I could pass a function to JSONX that would be invoked whenever an element is created.

The implementation might look like simply abstracting all `React.createElement` calls to a `jxmProvider(type, props, children)`, and probably making `(...args) => React.createElement(...args)` the default provider or something.

I put together something very similar to this utility for React, and this is a feature I was planning on adding, but I think it makes a lot of sense to go with something that already exists.

**Comment by yawetse** [comment]
that would be very cool, I'd love to see a proof of concept PR if you get the chance to tinker with it

**Comment by yawetse** [comment]
@Swivelgames  Do you have an example of an alternative jxmProvider? I think it's a pretty doable

**Comment by yawetse** [comment]
Closing as part of repository maintenance after #1088 landed on main with current dependency updates and passing tests/build. Please open a new issue with a minimal reproduction if this still reproduces on current main or the next published package.
